docs/comparisons.rst

```diff
@@ -0,0 +1,132 @@
+
+* Full statement source code presentation
+* Presentation of variables present in assertion statements
+
+Marbles goes beyond pytest with:
+
+* Note annotations
+* Custom assertion mixins
+
+Finally, marbles tests have the same structure as unittest tests,
+which are more explicit than pytest tests, and therefore are easier to
+learn and to reason about at failure time.
+
+If you're familiar with pytest, you'll probably find that writing
+marbles tests is more typing than you're used to, but we hope, no
+matter which tool you know well, responding to a test failure will be
+faster and easier with marbles. In short, pytest is write optimized,
+while marbles is read optimized.
+
+Similarities
+~~~~~~~~~~~~
+
+Assertion Source
+""""""""""""""""
+
+Both marbles and pytest present the :ref:`source code <source>` of the
+whole assertion statement that failed, which is more useful than a
+typical Python traceback.
+
+We believe both tools provide an equivalent benefit here.
+
+Local Variables
+"""""""""""""""
+
+Both marbles and pytest present some of the local variables present at
+the time an assertion in your test failed.
+
+While pytest presents just the variables involved in the assertion
+(and shows each sub-expression involved in the assertion), marbles
+presents all public local variables in scope in the test method.
+
+Marbles is a little more flexible here. While pytest has the ability
+to expand complex expressions and present all of their constituent
+names in the failure message, it doesn't present any other variables
+leading up to the assertion, which may provide valuable context.
+
+With marbles, you can get the same effect as pytest by simply
+decomposing a complex assertion statement into its constituent parts,
+and giving each of these parts a distinct local variable binding.
+
+Consider the following pytest code::
+
+    assert a * b < c * d
+
+If this assertion fails, pytest will show you the values of the
+expressions ``a * b`` and ``c * d``, as well as the individual values
+of each variable ``a``, ``b``, ``c``, and ``d``.
+
+In marbles, one could instead write this test as::
+
+    lhs = a * b
+    rhs = c * d
+    self.assertLess(lhs, rhs)
+
+Advantages of marbles
+~~~~~~~~~~~~~~~~~~~~~
+
+Note annotations
+""""""""""""""""
+
+Marbles gives the test author a place to :ref:`annotate assertion
+statements <notes>` with information about not just a description of
+what assertion failed, but also information about why that assertion
+is important, or maybe what the test consumer should do to handle that
+failure.
+
+Custom assertion mixins
+"""""""""""""""""""""""
+
+Pytest relies on the bare ``assert`` keyword, and encourages use of it
+directly. This puts the burden on the test consumer to understand the
+author's intent. As a consumer, you need to parse the logic of the
+assertion condition and read the rest of the test to understand what's
+going on.
+
+Marbles provides, through :mod:`marbles.mixins` classes, richly-named
```

Comment:
This section obscures one of our original choices, which was to use `unittest` *because* of its assertions and because we *don't* like tests with the bare `assert` keyword.